tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinLibSources.kt

```diff
@@ -115,15 +115,5 @@ fun kotlinComponent(module: String): ComponentIdentifier =
 
 
 private
-fun kotlinModuleVersionIdentifier(module: String): ModuleVersionIdentifier =
-    DefaultModuleVersionIdentifier("org.jetbrains.kotlin", module, embeddedKotlinVersion)
-
-
-private
-fun moduleId(group: String, module: String, version: String) =
-    object : ModuleComponentIdentifier {
-        override fun getGroup() = group
-        override fun getModule() = module
-        override fun getVersion() = version
-        override fun getDisplayName() = "$group:$module:$version"
-    }
+fun moduleId(group: String, module: String, version: String): ModuleComponentIdentifier =
```

Comment:
Isn't `version` nullable in Gradle? Maybe this is a different thing and I'm conflating the two.

Reply:
It can't be null in this context by definition.

Reply:
What is the `ModuleComponentIdentifier`?
The [javadocs](https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/component/ModuleComponentIdentifier.html) don't really do a good job explaining what it is.